### PR TITLE
portable: Cleaned up includes to only (and always) include what we use

### PR DESCRIPTION
**Pull request by GabrielRavier**
Some files fail to include files that they require (i.e. they depend on another include to indirectly include what they need), or include files that they don't need. This PR fixes this.

### CODE REVIEW

File: src/Backends/Rendering/Window/Software/GLFW3.cpp [hunk 1]
```diff
@@ -11,7 +11,6 @@
 #include <GLFW/glfw3.h>
 
 #include "../../../Misc.h"
-#include "../../../Shared/GLFW3.h"
```
Comment (Clownacy) [inline src/Backends/Rendering/Window/Software/GLFW3.cpp:14]: This is needed for the same reason any file includes its own header - to detect conflicting definitions. It's the same for the other shared headers.

File: src/Sound.cpp [hunk 2]
```diff
@@ -20,7 +20,6 @@ equivalents.
 #include "WindowsWrapper.h"
 
 #include "Backends/Audio.h"
-#include "Main.h"
```
Comment (Clownacy) [inline src/Sound.cpp:23]: This should be commented-out, as with the other unneeded header.

File: src/Backends/Audio/SDL2.cpp [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
 
 #include <stddef.h>
-#include <stdio.h>
 #include <string.h>
 #include <string>
 
 #include "SDL.h"
+#include "SDL_audio.h"
+#include "SDL_error.h"
+#include "SDL_stdinc.h"
```
Comment (Clownacy) [inline src/Backends/Audio/SDL2.cpp:10]: I'd still rather stick to `SDL.h`.

File: src/Backends/Rendering/Window/OpenGL3/GLFW3.cpp [hunk 4]
```diff
@@ -1,17 +1,13 @@
 #include "../OpenGL.h"
 
 #include <stddef.h>
-#include <stdlib.h>
 
-#ifdef USE_OPENGLES2
-#include <GLES2/gl2.h>
-#else
+#ifndef USE_OPENGLES2
```
Comment (Clownacy) [inline src/Backends/Rendering/Window/OpenGL3/GLFW3.cpp:5]: Isn't this needed so I can access the GLES2 functions?

File: src/Backends/Rendering/Window/Software/GLFW3.cpp (hunk 1 above)
Comment (GabrielRavier) [reply src/Backends/Rendering/Window/Software/GLFW3.cpp:14]: That's "its own header" ? Guess I'll restore it, then.

Comment (GabrielRavier) [reply src/Backends/Rendering/Window/Software/GLFW3.cpp]: Fixed

File: src/Backends/Rendering/Window/OpenGL3/GLFW3.cpp (hunk 4 above)
Comment (GabrielRavier) [reply src/Backends/Rendering/Window/OpenGL3/GLFW3.cpp:5]: What GLES2 functions ? I can't see any GLES2 functions being used in that file, can you ?

Comment (Clownacy) [reply src/Backends/Rendering/Window/OpenGL3/GLFW3.cpp:5]: GLES2 provides the OpenGL functions in OpenGL ES 2 builds, like how glad provides them in OpenGL3 builds.

File: src/Sound.cpp (hunk 2 above)
Comment (GabrielRavier) [reply src/Sound.cpp:23]: Fixed.

File: src/Backends/Audio/SDL2.cpp (hunk 3 above)
Comment (GabrielRavier) [reply src/Backends/Audio/SDL2.cpp]: Fixed.

File: src/Backends/Rendering/Window/OpenGL3/GLFW3.cpp (hunk 4 above)
Comment (GabrielRavier) [reply src/Backends/Rendering/Window/OpenGL3/GLFW3.cpp:5]: Yes, but what OpenGL functions are you talking about ? `src/Backends/Rendering/Window/OpenGL3/GLFW3.cpp` doesn't use any that I can see, it only conditionally uses `gladLoadGLLoader` and `GLAD_GL_VERSION_3_2` if OpenGLES 2 isn't enabled (using glad) and the glfw functions are universal afaik.

Comment (GabrielRavier) [reply src/Backends/Rendering/Window/OpenGL3/GLFW3.cpp:5]: I can build CSE2 with a configuration that uses GLFW3 and OpenGLES2 with no problems with this PR.

Comment (Clownacy) [reply src/Backends/Rendering/Window/OpenGL3/GLFW3.cpp:5]: Oh right. That's fine then.

**Pull request merged**